From 6549fb7405ca6f6d54cfac8a2a220a128908b527 Mon Sep 17 00:00:00 2001
From: sean meng <sean.meng@lairdtech.com>
Date: Tue, 22 Jan 2019 13:30:58 -0500
Subject: [PATCH] buildroot:GLib MainLoop is not supported.

---
 src/firewall/server/server.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/firewall/server/server.py b/src/firewall/server/server.py
index 9c7002a..1419b74 100644
--- a/src/firewall/server/server.py
+++ b/src/firewall/server/server.py
@@ -92,7 +92,7 @@ def run_server(debug_gc=False):
         name = dbus.service.BusName(config.dbus.DBUS_INTERFACE, bus=bus)
         service = FirewallD(name, config.dbus.DBUS_PATH)

-        mainloop = GLib.MainLoop()
+        mainloop = GObject.MainLoop()
         slip.dbus.service.set_mainloop(mainloop)
         if debug_gc:
             GLib.timeout_add_seconds(gc_timeout, gc_collect)
--
2.7.4

