diff --git a/parser/parser.c b/parser/parser.c
index acac406..9ae9448 100644
--- a/parser/parser.c
+++ b/parser/parser.c
@@ -324,6 +324,9 @@ static int parse_partitions(parsertype p, void *cfg, struct swupdate_cfg *swcfg)

 		partition->provided = 1;

+		if(!strlen(partition->device) && strlen(partition->path))
+			snprintf(partition->device, MAX_VOLNAME, "%s", partition->path);
+
 		if (!strlen(partition->volname) || !strlen(partition->device)) {
 			ERROR("Partition incompleted in description file");
 			free_image(partition);
--
2.7.4

