--- a/core/cpio_utils.c	2022-12-13 12:36:58.000000000 
+++ b/core/cpio_utils.c	2023-03-30 17:18:56.000000000 
@@ -138,12 +138,12 @@
 		if (ret < 0) {
 			if (errno == EINTR)
 				continue;
-			ERROR("cannot write %lu bytes: %s", len, strerror(errno));
+			ERROR("cannot write %zu bytes: %s", len, strerror(errno));
 			return -1;
 		}
 
 		if (ret == 0) {
-			ERROR("cannot write %lu bytes: %s", len, strerror(errno));
+			ERROR("cannot write %zu bytes: %s", len, strerror(errno));
 			return -1;
 		}
 
--- a/core/notifier.c	2022-12-13 12:36:58.000000000 
+++ b/core/notifier.c	2023-03-30 18:59:14.000000000 
@@ -492,7 +492,11 @@
 	if (sd_booted() && getenv("JOURNAL_STREAM") != NULL) {
 		dev_t device;
 		ino_t inode;
+#ifdef __USE_FILE_OFFSET64		
+		if (sscanf(getenv("JOURNAL_STREAM"), "%llu:%llu", &device, &inode) == 2) {
+#else
 		if (sscanf(getenv("JOURNAL_STREAM"), "%lu:%lu", &device, &inode) == 2) {
+#endif
 			struct stat statbuffer;
 			if (fstat(fileno(stderr), &statbuffer) == 0) {
 				if (inode == statbuffer.st_ino && device == statbuffer.st_dev) {
