--- a/handlers/ubivol_handler.c
+++ b/handlers/ubivol_handler.c
@@ -221,6 +221,12 @@
 	return err;
 }
 
+static bool check_ubi_skipcheck(struct img_type *img)
+{
+	return !strcmp(img->type_data, "static") &&
+ 		strtobool(dict_get_value(&img->properties, "skip-check"));
+}
+
 static int resize_volume(struct img_type *cfg, long long size)
 {
 	struct flash_description *nandubi = get_flash_info();
@@ -324,6 +330,7 @@
 		req.alignment = 1;
 		req.bytes = size;
 		req.name = cfg->volname;
+		req.flags = check_ubi_skipcheck(cfg) ? UBI_VOL_SKIP_CRC_CHECK_FLG : 0;
 		err = ubi_mkvol(nandubi->libubi, node, &req);
 		if (err < 0) {
 			ERROR("cannot create %s UBI volume %s of %lld bytes",
