--- a/parser/parser.c	2023-03-30 20:19:35.000000000 
+++ b/parser/parser.c	2023-03-30 21:24:25.000000000 
@@ -264,8 +264,8 @@
 
 	setting = find_node(p, cfg, "hardware-compatibility", swcfg);
 	if (setting == NULL) {
-		ERROR("HW compatibility not found");
-		return -1;
+		WARN("HW compatibility not found");
+		return 0;
 	}
 
 	count = get_array_length(p, setting);
--- a/core/hw-compatibility.c	2022-12-13 12:36:58.000000000 
+++ b/core/hw-compatibility.c	2022-12-13 12:36:58.000000000 
@@ -22,6 +22,9 @@
 {
 	struct hw_type *hw;
 	int ret;
+
+	if (LIST_EMPTY(&cfg->hardware))
+		return 0;
 
 	ret = get_hw_revision(&cfg->hw);
 	if (ret < 0)
