diff --git a/Makefile b/Makefile
index e9b417f..4dc6dfe 100644
--- a/Makefile
+++ b/Makefile
@@ -38,6 +38,7 @@ all: all_noverify verify
 
 all_noverify: $(LIBREG) crda intersect regdbdump db2rd optimize
 
+ifneq ($(CRDA_NO_SIGN_CHECK),y)
 ifeq ($(USE_OPENSSL),1)
 CFLAGS += -DUSE_OPENSSL -DPUBKEY_DIR=\"$(RUNTIME_PUBKEY_DIR)\" `pkg-config --cflags openssl`
 LDLIBS += `pkg-config --libs openssl`
@@ -51,6 +52,7 @@ LDLIBS += -lgcrypt
 $(LIBREG): keys-gcrypt.c
 
 endif
+endif
 MKDIR ?= mkdir -p
 INSTALL ?= install
 
