From 2a361e197285f01eaf6a5195a726f040252b4b6f Mon Sep 17 00:00:00 2001
From: Ben Whitten <ben.whitten@lairdtech.com>
Date: Thu, 11 Aug 2016 15:40:06 +0100
Subject: [PATCH] Adjust reset the GPIO to the WB50 platform

---
 reset_lgw.sh | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/reset_lgw.sh b/reset_lgw.sh
index 64cb3d1..b3dfb82 100755
--- a/reset_lgw.sh
+++ b/reset_lgw.sh
@@ -1,38 +1,38 @@
 #!/bin/sh
 
-# This script is intended to be used on IoT Starter Kit platform only, it
+# This script is intended to be used on WB50 platform only, it
 # performs the following actions:
-#       - export/unpexort GPIO7 used to reset the SX1301 chip
+#       - export/unpexort GPIO26 used to reset the SX1301 chip
 #
 # Usage examples:
 #       ./reset_lgw.sh stop
 #       ./reset_lgw.sh start
 
-# The reset pin of SX1301 is wired with RPi GPIO7
-IOT_SK_SX1301_RESET_PIN=7
+# The reset pin of SX1301 is wired with WB50 GPIO26
+IOT_SK_SX1301_RESET_PIN=26
 
 WAIT_GPIO() {
-    sleep 0.1
+    usleep 100000
 }
 
 iot_sk_init() {
-    # setup GPIO 7
+    # setup GPIO 26
     echo "$IOT_SK_SX1301_RESET_PIN" > /sys/class/gpio/export; WAIT_GPIO
 
-    # set GPIO 7 as output
-    echo "out" > /sys/class/gpio/gpio$IOT_SK_SX1301_RESET_PIN/direction; WAIT_GPIO
+    # set GPIO 26 as output
+    echo "out" > /sys/class/gpio/pioA$IOT_SK_SX1301_RESET_PIN/direction; WAIT_GPIO
 
     # write output for SX1301 reset
-    echo "1" > /sys/class/gpio/gpio$IOT_SK_SX1301_RESET_PIN/value; WAIT_GPIO
-    echo "0" > /sys/class/gpio/gpio$IOT_SK_SX1301_RESET_PIN/value; WAIT_GPIO
+    echo "1" > /sys/class/gpio/pioA$IOT_SK_SX1301_RESET_PIN/value; WAIT_GPIO
+    echo "0" > /sys/class/gpio/pioA$IOT_SK_SX1301_RESET_PIN/value; WAIT_GPIO
 
-    # set GPIO 7 as input
-    echo "in" > /sys/class/gpio/gpio$IOT_SK_SX1301_RESET_PIN/direction; WAIT_GPIO
+    # set GPIO 26 as input
+    echo "in" > /sys/class/gpio/pioA$IOT_SK_SX1301_RESET_PIN/direction; WAIT_GPIO
 }
 
 iot_sk_term() {
-    # cleanup GPIO 7
-    if [ -d /sys/class/gpio/gpio$IOT_SK_SX1301_RESET_PIN ]
+    # cleanup GPIO 26
+    if [ -d /sys/class/gpio/pioA$IOT_SK_SX1301_RESET_PIN ]
     then
         echo "$IOT_SK_SX1301_RESET_PIN" > /sys/class/gpio/unexport; WAIT_GPIO
     fi
-- 
2.7.4

