From ba3dc3f534e0a6309efe4d271abd6620dae7fd62 Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Mon, 15 Jul 2013 06:32:58 -0700
Subject: [PATCH 4/5] Ignore cscope.out files

---
 .gitignore |    1 +
 1 file changed, 1 insertion(+)

diff --git a/.gitignore b/.gitignore
index d78652e..65e28f6 100644
--- a/.gitignore
+++ b/.gitignore
@@ -21,3 +21,4 @@ binaries
 *.o
 tags
 *.swp
+cscope.out
\ No newline at end of file
-- 
1.7.9.5

