From 4131f16f39c8c40ad3b1b28c45e619a50e4fae9b Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Wed, 12 Feb 2014 20:37:05 -0800
Subject: [PATCH] Update version to Laird14

---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index aa2e62c..d566026 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird13
+EXTRAVERSION = -laird14
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.7.9.5

