From 57fe5f546094f82714d7e2109d4e6d81f2c8562d Mon Sep 17 00:00:00 2001
From: "zachary.coffy" <zachary.coffy@lairdtech.com>
Date: Fri, 30 May 2014 15:14:56 -0400
Subject: [PATCH] Update version to Laird19

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5be4a9b..40c9fca 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird18
+EXTRAVERSION = -laird19
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.8.3.2

