From 85889fcd0a6417ffaaa3e23f923ab85139043da4 Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Thu, 1 Aug 2013 14:30:15 -0700
Subject: [PATCH] Update wb45n kernel version to 3.8.0-laird3

---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6d8a3e9..467a375 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird2
+EXTRAVERSION = -laird3
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.7.9.5

