From ae4633fe28bb431554d1ce5628435f15f9d63a3e Mon Sep 17 00:00:00 2001
From: "zachary.coffy" <zachary.coffy@lairdtech.com>
Date: Wed, 11 Jun 2014 15:53:46 -0400
Subject: [PATCH] Update version to Laird20

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 40c9fca..a80a8a5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird19
+EXTRAVERSION = -laird20
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.8.3.2

