From 8240a5f0172577e7948a4a3fba20e8de5dccb981 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 17 Feb 2016 16:52:21 -0800
Subject: [PATCH] laird: update version to laird08

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6615637..6ca9b8c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird07
+EXTRAVERSION = -laird08
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

