From 1c0bc43eefa4ec785d76cc202a5397cad6da6493 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 17 Feb 2016 20:18:08 -0800
Subject: [PATCH] laird: update version to laird09

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6ca9b8c..d35c433 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird08
+EXTRAVERSION = -laird09
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

