From 1408fb8cbd6a33c528ef9b0be45c25ec98a25e15 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 25 Nov 2015 19:07:54 -0800
Subject: [PATCH] laird: update version to laird01

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d5d229d..756fc35 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION =
+EXTRAVERSION = -laird01
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

