From 4bc8e15cf6ae2060133e5b54af6a447728bd6743 Mon Sep 17 00:00:00 2001
From: Zachary Coffy <Zachary.Coffy@lairdtech.com>
Date: Wed, 2 Dec 2015 13:34:14 -0700
Subject: [PATCH] laird: update version to laird02

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 756fc35..087da82 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird01
+EXTRAVERSION = -laird02
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.8.3.1

