From b2edc441b1ef7d44a1b0a78eff1cf90d0311f345 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 28 Jan 2016 18:18:52 -0800
Subject: [PATCH] laird: update version to laird05

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 45c4723..9660fa7 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird04
+EXTRAVERSION = -laird05
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

