From 09264d465d87e05aac3b1038fd8ef21f63a18d4b Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 25 Feb 2016 09:11:42 -0800
Subject: [PATCH] laird: update version to laird10

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d35c433..0242915 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird09
+EXTRAVERSION = -laird10
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

