From 3ad4f9bf9fd11b2c099e54cfd8dbe48077151aec Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 3 Feb 2016 16:42:53 -0800
Subject: [PATCH] laird: update version to laird06

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index aef5ffb..013178b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird03
+EXTRAVERSION = -laird06
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

