From e509449c6eb3b73118a438007ba596f4e2688974 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Sat, 9 Apr 2016 11:38:34 -0700
Subject: [PATCH] laird: update version to laird13

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f28497d..f07d35a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird12
+EXTRAVERSION = -laird13
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

