From 99c523d06b810731b82fa9f5eb9f9e55303e8b44 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 9 Mar 2016 16:09:49 -0800
Subject: [PATCH] laird: update version to laird12

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 196ec16..f28497d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird11
+EXTRAVERSION = -laird12
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

