From 5d86e81e4c26164b171853a33fe65002a08c0592 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 3 Mar 2016 17:25:23 -0800
Subject: [PATCH] laird: update version to laird11

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0242915..196ec16 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 1
 SUBLEVEL = 13
-EXTRAVERSION = -laird10
+EXTRAVERSION = -laird11
 NAME = Series 4800
 
 # *DOCUMENTATION*
-- 
1.9.1

