From 514552308585dffd04d54b8e8b1810d5b35b4130 Mon Sep 17 00:00:00 2001
From: Adam Ruehl <adam.ruehl@lairdtech.com>
Date: Sun, 2 Sep 2018 21:02:34 -0400
Subject: [PATCH 2/5] Fix compiler errors.

---
 microxml.h    |  2 +-
 mxml-string.c | 16 ++++++++++++++++
 2 files changed, 17 insertions(+), 1 deletion(-)

diff --git a/microxml.h b/microxml.h
index aca170d..a5e8097 100644
--- a/microxml.h
+++ b/microxml.h
@@ -23,7 +23,7 @@
 /*
  * Include necessary headers...
  */
-
+#  include <stdarg.h>
 #  include <stdio.h>
 #  include <stdlib.h>
 #  include <string.h>
diff --git a/mxml-string.c b/mxml-string.c
index 62e2aa8..d6878da 100644
--- a/mxml-string.c
+++ b/mxml-string.c
@@ -27,6 +27,22 @@
 
 #include "config.h"
 
+_mxml_strdup(const char *s);
+_mxml_snprintf(char       *buffer,	/* I - Output buffer */
+               size_t     bufsize,	/* I - Size of output buffer */
+	       const char *format,	/* I - Printf-style format string */
+	       ...);
+char *					/* O - New string pointer */
+_mxml_strdupf(const char *format,	/* I - Printf-style format string */
+              ...);
+int					/* O - Number of bytes formatted */
+_mxml_vsnprintf(char       *buffer,	/* O - Output buffer */
+                size_t     bufsize,	/* O - Size of output buffer */
+		const char *format,	/* I - Printf-style format string */
+ 		va_list    ap);
+char *					/* O - New string pointer */
+_mxml_vstrdupf(const char *format,	/* I - Printf-style format string */
+               va_list    ap);		/* I - Pointer to additional arguments */
 
 /*
  * The va_copy macro is part of C99, but many compilers don't implement it.
-- 
2.7.4

