From ef63cb0eb0b84f89c928d88c21f5305061c2b5b8 Mon Sep 17 00:00:00 2001
From: Adam Ruehl <adam.ruehl@lairdtech.com>
Date: Sun, 2 Sep 2018 21:36:59 -0400
Subject: [PATCH 4/5] Include all necessary includes

---
 mxml-string.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/mxml-string.c b/mxml-string.c
index d6878da..27f46bc 100644
--- a/mxml-string.c
+++ b/mxml-string.c
@@ -27,8 +27,14 @@
 
 #include "config.h"
 
-_mxml_strdup(const char *s);
-_mxml_snprintf(char       *buffer,	/* I - Output buffer */
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <string.h>
+
+void _mxml_strdup(const char *s);
+void _mxml_snprintf(char       *buffer,	/* I - Output buffer */
                size_t     bufsize,	/* I - Size of output buffer */
 	       const char *format,	/* I - Printf-style format string */
 	       ...);
-- 
2.7.4

