From 115482f1ed306fbf9670cc62c1858353d3b14f16 Mon Sep 17 00:00:00 2001
From: Adam Ruehl <adam.ruehl@lairdtech.com>
Date: Sun, 2 Sep 2018 21:58:56 -0400
Subject: [PATCH 5/5] Fix compiler warnings.

---
 microxml.h    | 20 +++++++++++++++++---
 mxml-string.c | 24 +-----------------------
 2 files changed, 18 insertions(+), 26 deletions(-)

diff --git a/microxml.h b/microxml.h
index a5e8097..763e28d 100644
--- a/microxml.h
+++ b/microxml.h
@@ -30,7 +30,6 @@
 #  include <ctype.h>
 #  include <errno.h>
 
-
 /*
  * Constants...
  */
@@ -175,7 +174,6 @@ typedef const char *(*mxml_save_cb_t)(mxml_node_t *, int);
 typedef void (*mxml_sax_cb_t)(mxml_node_t *, mxml_sax_event_t, void *);  
 					/**** SAX callback function ****/
 
-
 /*
  * C++ support...
  */
@@ -188,6 +186,23 @@ extern "C" {
  * Prototypes...
  */
 
+void _mxml_strdup(const char *s);
+void _mxml_snprintf(char       *buffer,	/* I - Output buffer */
+               size_t     bufsize,	/* I - Size of output buffer */
+	       const char *format,	/* I - Printf-style format string */
+	       ...);
+char *					/* O - New string pointer */
+_mxml_strdupf(const char *format,	/* I - Printf-style format string */
+              ...);
+int					/* O - Number of bytes formatted */
+_mxml_vsnprintf(char       *buffer,	/* O - Output buffer */
+                size_t     bufsize,	/* O - Size of output buffer */
+		const char *format,	/* I - Printf-style format string */
+		va_list    ap);
+char *					/* O - New string pointer */
+_mxml_vstrdupf(const char *format,	/* I - Printf-style format string */
+               va_list    ap);		/* I - Pointer to additional arguments */
+
 extern void		mxmlAdd(mxml_node_t *parent, int where,
 			        mxml_node_t *child, mxml_node_t *node);
 extern void		mxmlDelete(mxml_node_t *node);
@@ -315,7 +330,6 @@ extern mxml_type_t	mxml_integer_cb(mxml_node_t *node);
 extern mxml_type_t	mxml_opaque_cb(mxml_node_t *node);
 extern mxml_type_t	mxml_real_cb(mxml_node_t *node);
 
-
 /*
  * C++ support...
  */
diff --git a/mxml-string.c b/mxml-string.c
index 27f46bc..a0a4e9b 100644
--- a/mxml-string.c
+++ b/mxml-string.c
@@ -26,29 +26,7 @@
  */
 
 #include "config.h"
-
-#include <stdio.h>
-#include <stdlib.h>
-#include <stdarg.h>
-#include <stddef.h>
-#include <string.h>
-
-void _mxml_strdup(const char *s);
-void _mxml_snprintf(char       *buffer,	/* I - Output buffer */
-               size_t     bufsize,	/* I - Size of output buffer */
-	       const char *format,	/* I - Printf-style format string */
-	       ...);
-char *					/* O - New string pointer */
-_mxml_strdupf(const char *format,	/* I - Printf-style format string */
-              ...);
-int					/* O - Number of bytes formatted */
-_mxml_vsnprintf(char       *buffer,	/* O - Output buffer */
-                size_t     bufsize,	/* O - Size of output buffer */
-		const char *format,	/* I - Printf-style format string */
- 		va_list    ap);
-char *					/* O - New string pointer */
-_mxml_vstrdupf(const char *format,	/* I - Printf-style format string */
-               va_list    ap);		/* I - Pointer to additional arguments */
+#include "microxml.h"
 
 /*
  * The va_copy macro is part of C99, but many compilers don't implement it.
-- 
2.7.4

