--- a/src/v4l_id/v4l_id.c	2021-10-16 17:31:03.000000000 
+++ b/src/v4l_id/v4l_id.c	2022-03-31 13:40:37.000000000 
@@ -71,9 +71,11 @@
                 printf("ID_V4L_VERSION=2\n");
                 printf("ID_V4L_PRODUCT=%s\n", v2cap.card);
                 printf("ID_V4L_CAPABILITIES=:");
+#ifdef V4L2_CAP_DEVICE_CAPS
                 if (v2cap.capabilities & V4L2_CAP_DEVICE_CAPS)
                         capabilities = v2cap.device_caps;
                 else
+#endif                
                         capabilities = v2cap.capabilities;
                 if ((capabilities & V4L2_CAP_VIDEO_CAPTURE) > 0 ||
                     (capabilities & V4L2_CAP_VIDEO_CAPTURE_MPLANE) > 0)
