From 558122f7f12357e10ab43729a88be7a2bd3128e6 Mon Sep 17 00:00:00 2001
From: Dimitrios Siganos <dimitris@siganos.org>
Date: Thu, 18 Jul 2013 17:52:24 +0100
Subject: [PATCH] Setting wb45n kernel version to 3.8.0-laird2

---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index c204b89..6d8a3e9 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird1
+EXTRAVERSION = -laird2
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.7.9.5

