From c881341d0f1e5d338034eb16dd47bf0cb579cd06 Mon Sep 17 00:00:00 2001
From: "zachary.coffy" <zachary.coffy@lairdtech.com>
Date: Wed, 18 Jun 2014 16:00:47 -0400
Subject: [PATCH] Update version to Laird21

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a80a8a5..f6742e3 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird20
+EXTRAVERSION = -laird21
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.8.3.2

