From 4492130c8a0a9de60c4cd000f219d00421f91c66 Mon Sep 17 00:00:00 2001
From: "zachary.coffy" <zachary.coffy@lairdtech.com>
Date: Wed, 14 May 2014 12:39:54 -0400
Subject: [PATCH] Update version to Laird17

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e21660d..061d1b4 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird16
+EXTRAVERSION = -laird17
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.8.3.2

