From cd13b3430282d6168422dd43e308e40c45420fa1 Mon Sep 17 00:00:00 2001
From: Zachary Coffy <Zachary.Coffy@lairdtech.com>
Date: Wed, 30 Oct 2013 12:41:53 -0400
Subject: [PATCH] Adjust version for release laird8

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4bba390..9788cc5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird7
+EXTRAVERSION = -laird8
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.8.3.msysgit.0

