From 2b1897d0eaa42da19f85bfaf87ce93e4409141f3 Mon Sep 17 00:00:00 2001
From: Zachary Coffy <Zachary.Coffy@lairdtech.com>
Date: Wed, 24 Sep 2014 13:15:49 -0600
Subject: [PATCH] Update version to Laird24

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 9159eab..d5b3f77 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird23
+EXTRAVERSION = -laird24
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.8.3.1

