From 56006dfcb3bf489495c2b11baac25db5af1aca0c Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Wed, 15 Jan 2014 14:28:40 -0500
Subject: [PATCH] Update version to laird12

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d558276..64c34ff 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 8
 SUBLEVEL = 0
-EXTRAVERSION = -laird11
+EXTRAVERSION = -laird12
 NAME = Unicycling Gorilla
 
 # *DOCUMENTATION*
-- 
1.8.1.2

