diff -uar dbus-1.10.16-orig/dbus/dbus-message.c dbus-1.10.16-new/dbus/dbus-message.c
--- dbus-1.10.16-orig/dbus/dbus-message.c	2016-10-07 16:27:14.000000000 -0400
+++ dbus-1.10.16-new/dbus/dbus-message.c	2017-07-05 17:00:41.175134864 -0400
@@ -4327,7 +4327,8 @@
       goto failed;
     }
 
-  _dbus_string_delete (&loader->data, 0, header_len + body_len);
+  /* Laird: clear (zero) the message/creds */
+  _dbus_string_zero_delete (&loader->data, 0, header_len + body_len);
 
   /* don't waste more than 2k of memory */
   _dbus_string_compact (&loader->data, 2048);
diff -uar dbus-1.10.16-orig/dbus/dbus-string.c dbus-1.10.16-new/dbus/dbus-string.c
--- dbus-1.10.16-orig/dbus/dbus-string.c	2015-09-30 10:48:40.000000000 -0400
+++ dbus-1.10.16-new/dbus/dbus-string.c	2017-07-05 17:00:41.175134864 -0400
@@ -1203,6 +1203,50 @@
   delete (real, start, len);
 }
 
+/* Laird: clear (zero) message/creds */
+/* Laird: same as delete() but zeros the end of string (len bytes) */
+static void
+zero_delete (DBusRealString *real,
+        int             start,
+        int             len)
+{
+  if (len == 0)
+    return;
+
+  memmove (real->str + start, real->str + start + len, real->len - (start + len));
+  real->len -= len;
+  /* Laird: zero the end of the string */
+  memset (real->str + real->len, '\0', len);
+  real->str[real->len] = '\0';
+}
+
+/* Laird: clear (zero) message/creds */
+/* Laird: same as _dbus_string_delete() but zeros end of string (len bytes) */
+/**
+ * Deletes a segment of a DBusString with length len starting at
+ * start. (Hint: to clear an entire string, setting length to 0
+ * with _dbus_string_set_length() is easier.)
+ *
+ * Laird: This variant zeros the end of the string.
+ *
+ * @param str the DBusString
+ * @param start where to start deleting
+ * @param len the number of bytes to delete
+ */
+void
+_dbus_string_zero_delete (DBusString       *str,
+                        int               start,
+                        int               len)
+{
+  DBUS_STRING_PREAMBLE (str);
+  _dbus_assert (start >= 0);
+  _dbus_assert (len >= 0);
+  _dbus_assert (start <= real->len);
+  _dbus_assert (len <= real->len - start);
+
+  zero_delete (real, start, len);
+}
+
 static dbus_bool_t
 copy (DBusRealString *source,
       int             start,
diff -uar dbus-1.10.16-orig/dbus/dbus-string.h dbus-1.10.16-new/dbus/dbus-string.h
--- dbus-1.10.16-orig/dbus/dbus-string.h	2015-09-30 10:48:40.000000000 -0400
+++ dbus-1.10.16-new/dbus/dbus-string.h	2017-07-05 17:09:55.379134864 -0400
@@ -220,6 +220,10 @@
 void          _dbus_string_delete                (DBusString        *str,
                                                   int                start,
                                                   int                len);
+/* Laird: clear (zero) message/creds */
+void          _dbus_string_zero_delete           (DBusString        *str,
+                                                  int                start,
+                                                  int                len);
 DBUS_PRIVATE_EXPORT
 dbus_bool_t   _dbus_string_move                  (DBusString        *source,
                                                   int                start,
