From c82c32a97672431252078dea9e0a21076638595f Mon Sep 17 00:00:00 2001
From: Joe Conley <joe.conley@lairdtech.com>
Date: Wed, 22 Mar 2017 16:52:40 -0400
Subject: [PATCH] Update version to 4.4.39-laird07.02

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7f01c4f..8197e4f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 39
-EXTRAVERSION = -laird07.01
+EXTRAVERSION = -laird07.02
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
-- 
2.7.4

