diff -Naur a/setup.py b/setup.py
--- a/setup.py	2021-04-16 16:13:47.000000000 +0000
+++ b/setup.py	2022-02-10 14:35:21.529190274 +0000
@@ -19,7 +19,7 @@
         commands =  [
                     ["autoconf"],
                     ["autoheader"],
-                    ["./configure", "--without-ecc"],
+                    ["./configure", "--host=x86_64", "--without-ecc"],
                     ]
         if not os.path.exists(os.path.join(os.path.dirname(__file__), 'DTLSSocket','tinydtls','dtls.c')):
             run_command(["git", "submodule", "update", "--init"])
