--- a/jitterentropy.c	2021-06-16 18:41:18.000000000 
+++ b/jitterentropy.c	2021-06-16 18:42:10.000000000 
@@ -121,13 +121,8 @@
 
 #include <linux/module.h>
 
-static bool fips_fail_jent_apt_insert = 0;
-module_param(fips_fail_jent_apt_insert, bool, 0444);
-MODULE_PARM_DESC(fips_fail_jent_apt_insert, "FIPS testing: force jent_apt_insert() health test to fail.");
-
-static bool fips_fail_jent_rct_insert = 0;
-module_param(fips_fail_jent_rct_insert, bool, 0444);
-MODULE_PARM_DESC(fips_fail_jent_rct_insert, "FIPS testing: force jent_rct_insert() health test to fail.");
+static __u32 fips_fail_jent_apt_insert = 0;
+static __u32 fips_fail_jent_rct_insert = 0;
 
 /***************************************************************************
  * Adaptive Proportion Test
