--- a/src/libsystemd/sd-journal/sd-journal.c	2022-03-11 03:33:33.000000000 
+++ b/src/libsystemd/sd-journal/sd-journal.c	2022-04-05 03:38:35.000000000 
@@ -1276,6 +1276,9 @@
         assert(fd >= 0 || path);
 
         if (fd < 0) {
+                if (!path)
+                        goto finish;
+
                 if (j->toplevel_fd >= 0)
                         /* If there's a top-level fd defined make the path relative, explicitly, since otherwise
                          * openat() ignores the first argument. */
