diff --git a/src/shared/mount-util.h b/src/shared/mount-util.h
index 8b07611ec8..d665d21e3a 100644
--- a/src/shared/mount-util.h
+++ b/src/shared/mount-util.h
@@ -27,8 +27,9 @@
 #define TMPFS_LIMITS_SYS_FS_CGROUP   TMPFS_LIMITS_EMPTY_OR_ALMOST
 
 /* On an extremely small device with only 256MB of RAM, 20% of RAM should be enough for the re-execution of
- * PID1 because 16MB of free space is required. */
-#define TMPFS_LIMITS_RUN             ",size=20%,nr_inodes=800k"
+ * PID1 because 16MB of free space is required.
+ * Laird devices could have significantly less than the assumed 256MB, increase to 30% */
+#define TMPFS_LIMITS_RUN             ",size=30%,nr_inodes=800k"
 
 /* The limit used for various nested tmpfs mounts, in particular for guests started by systemd-nspawn.
  * 10% of RAM (using 16GB of RAM as a baseline) translates to 400k inodes (assuming 4k each) and 25%
