From 3be38146d3f31d2fe51ebf67f6b5b4ba6c68bee1 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Fri, 27 May 2016 13:26:33 -0700
Subject: [PATCH] laird: update version to laird01

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index aad8627..058d2d2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 11
-EXTRAVERSION =
+EXTRAVERSION = -laird01
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
-- 
1.9.1

