From ab2debe60546c0924cd6d48f33ee44997b832ab9 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 10 Nov 2016 17:05:36 -0800
Subject: [PATCH] laird: Update version to 4.4.11-laird07

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6d887d3..564adb2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 11
-EXTRAVERSION = -laird06
+EXTRAVERSION = -laird07
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
-- 
1.9.1

