From d654a7aeb95b44fef2843466d57688d23809b36e Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 8 Sep 2016 16:03:28 -0700
Subject: [PATCH] laird: Update version to 4.4.11-laird07.01

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 058d2d2..1d7883c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 11
-EXTRAVERSION = -laird01
+EXTRAVERSION = -laird07.01
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
-- 
1.9.1

