From 1f49f8b5886cc44c1f0cf776603e08c5ec2e0ea6 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 1 Sep 2016 15:43:41 -0700
Subject: [PATCH] laird: Update version to 4.4.11-laird05

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ecf530e..9f057e24 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 11
-EXTRAVERSION = -laird04
+EXTRAVERSION = -laird05
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
-- 
1.9.1

