From 7eee5e4194d3b47a90b43a6243812195d424550b Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Mon, 28 Nov 2016 13:42:42 -0800
Subject: [PATCH] Update version to 4.4.11-laird07.02

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1d7883c..f3ed62b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 4
 PATCHLEVEL = 4
 SUBLEVEL = 11
-EXTRAVERSION = -laird07.01
+EXTRAVERSION = -laird07.02
 NAME = Blurry Fish Butt
 
 # *DOCUMENTATION*
-- 
1.9.1

