From 1e616ca2167be14ba5e44929a743ba7cd791e3ca Mon Sep 17 00:00:00 2001
From: sean meng <sean.meng@lairdtech.com>
Date: Mon, 11 Feb 2019 10:30:13 -0500
Subject: [PATCH] fix-openssl-version

---
 openbsd-compat/openssl-compat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/openbsd-compat/openssl-compat.c b/openbsd-compat/openssl-compat.c
index 8b4a362..e332b0e 100644
--- a/openbsd-compat/openssl-compat.c
+++ b/openbsd-compat/openssl-compat.c
@@ -76,7 +76,7 @@ ssh_OpenSSL_add_all_algorithms(void)
 	ENGINE_load_builtin_engines();
 	ENGINE_register_all_complete();
 
-#if OPENSSL_VERSION_NUMBER < 0x10001000L
+#if OPENSSL_VERSION_NUMBER < 0x10101000L
 	OPENSSL_config(NULL);
 #else
 	OPENSSL_init_crypto(OPENSSL_INIT_ADD_ALL_CIPHERS |
-- 
2.7.4

