From ae0a4f27a76cd190859df959250d238b3b9f4b36 Mon Sep 17 00:00:00 2001
From: Steve deRosier <Steve.Derosier@lairdtech.com>
Date: Mon, 8 Jul 2013 15:51:10 -0700
Subject: [PATCH 3/4] Save default env

---
 common/env_common.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/common/env_common.c b/common/env_common.c
index 906b41f..19161e5 100644
--- a/common/env_common.c
+++ b/common/env_common.c
@@ -141,6 +141,7 @@ void set_default_env(const char *s)
 		error("Environment import failed: errno = %d\n", errno);
 
 	gd->flags |= GD_FLG_ENV_READY;
+	saveenv();
 }
 
 
-- 
1.7.9.5

