From c9d1ca114ccb631c3f62e5d5d8194418b759857e Mon Sep 17 00:00:00 2001
From: "Doug.Smith" <doug.smith@lairdconnect.com>
Date: Thu, 7 Jul 2022 13:48:24 -0400
Subject: [PATCH] Reproducible build: do not include build date

---
 util/mkbuildinf.pl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/util/mkbuildinf.pl b/util/mkbuildinf.pl
index 1c27387..7af0edc 100755
--- a/util/mkbuildinf.pl
+++ b/util/mkbuildinf.pl
@@ -28,7 +28,7 @@ print <<"END_OUTPUT";
  */
 
 #define PLATFORM "platform: $platform"
-#define DATE "built on: $date"
+#define DATE "built on: reproducible build, date unspecified"
 
 /*
  * Generate compiler_flags as an array of individual characters. This is a
-- 
2.17.1

