--- a/Makefile
+++ b/Makefile
@@ -1,26 +1,25 @@
 TARGET = mint
 OBJECTS = mint.o ipv4_net.o ipv6_net.o transmit.o  log.o
-CFLAGS = -W -Wall -O3 -g
-LDFLAGS =
+CFLAGS += -W -Wall -O3 -g
 SRCDIR = src/
 
 all:	$(OBJECTS)	
-	gcc $(LDFLAGS) -o $(TARGET) $(OBJECTS)
+	$(CC) $(LDFLAGS) -o $(TARGET) $(OBJECTS)
 
 ipv4_net.o:	$(SRCDIR)ipv4_net.c
-	gcc $(CFLAGS) $(SRCDIR)ipv4_net.c -c
+	$(CC) $(CFLAGS) $(SRCDIR)ipv4_net.c -c
 
 ipv6_net.o:	$(SRCDIR)ipv6_net.c
-	gcc $(CFLAGS) $(SRCDIR)ipv6_net.c -c
+	$(CC) $(CFLAGS) $(SRCDIR)ipv6_net.c -c
 
 transmit.o:	$(SRCDIR)transmit.c
-	gcc $(CFLAGS) $(SRCDIR)transmit.c -c
+	$(CC) $(CFLAGS) $(SRCDIR)transmit.c -c
 	
 mint.o:		$(SRCDIR)mint.c
-	gcc $(CFLAGS) $(SRCDIR)mint.c -c
+	$(CC) $(CFLAGS) $(SRCDIR)mint.c -c
 
 log.o:		$(SRCDIR)log.c
-	gcc $(CFLAGS) $(SRCDIR)log.c -c
+	$(CC) $(CFLAGS) $(SRCDIR)log.c -c
 
 tags:
 	find ./ -name '*.[ch]' | xargs ctags -a
