diff --git a/src/ipv6_net.c b/src/ipv6_net.c
--- a/src/ipv6_net.c
+++ b/src/ipv6_net.c
@@ -37,11 +37,7 @@ static int host2addr(char *ip6_host, struct in6_addr *group6_addr)
 		memcpy(group6_addr, hp->h_addr_list[0], 16);
 		}
 	
-#if !defined(linux)
 	if (!IN6_IS_ADDR_MULTICAST(group6_addr))
-#else	
-	if (!IN6_IS_ADDR_MULTICAST(group6_addr->in6_u.u6_addr8))
-#endif
 		{
 		log( "IPv6: %s address is not a multicast IPv6 address.\n", ip6_host);
 		return -ENOMULTI;
