diff -Naur a/src/mint.c b/src/mint.c
--- a/src/mint.c	2022-02-03 15:44:32.000000000 -0500
+++ b/src/mint.c	2022-02-03 15:46:30.967825903 -0500
@@ -67,6 +67,13 @@
 	not_logging = 0, send_bytes=64;
 FILE *loggerf;
 
+/*
+ * Address length may not be more than MAX_ADDR_LEN.
+ * Even in the case of IPv6.
+ */
+#define MAX_ADDR_LEN	45 
+char hostname[MAX_ADDR_LEN+1];
+
 /* 
  * Filename path may not be more than PATH_LEN.
  */
diff -Naur a/src/mint.h b/src/mint.h
--- a/src/mint.h	2022-02-03 15:44:32.000000000 -0500
+++ b/src/mint.h	2022-02-03 15:46:14.147924188 -0500
@@ -2,13 +2,6 @@
 		port_num, ttl_val, qos_val,
 		num_send, send_mcast, ipv6_sup, send_bytes;
 
-/*
- * Address length may not be more than MAX_ADDR_LEN.
- * Even in the case of IPv6.
- */
-#define MAX_ADDR_LEN	45 
-char hostname[MAX_ADDR_LEN+1];
-
 int mc_sender(char *, int, int, u_char, int, int,int);
 int mc_receiver(char *, int, int);
 
