--- host-openssl-1.0.2.orig/ssl/s3_clnt.c	2015-05-11 20:21:16.998081416 -0400
+++ host-openssl-1.0.2/ssl/s3_clnt.c	2015-05-11 20:22:31.437081407 -0400
@@ -963,6 +963,7 @@ int ssl3_get_server_hello(SSL *s)
                                      s->tls_session_secret_cb_arg)) {
             s->session->cipher = pref_cipher ?
                 pref_cipher : ssl_get_cipher_by_char(s, p + j);
+            s->s3->flags |= SSL3_FLAGS_CCS_OK;
             s->hit = 1;
         }
     }
