From 6d255ce7cad11eaa07ff9b8087fc117dae8b2a86 Mon Sep 17 00:00:00 2001
From: sean meng <sean.meng@lairdtech.com>
Date: Wed, 30 Jan 2019 16:30:18 -0500
Subject: [PATCH] disable cairo

---
 tests/scanner/regress.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/scanner/regress.h b/tests/scanner/regress.h
index e1c995a..d9d286b 100644
--- a/tests/scanner/regress.h
+++ b/tests/scanner/regress.h
@@ -1,6 +1,8 @@
 #ifndef __GITESTTYPES_H__
 #define __GITESTTYPES_H__
 
+#define _GI_DISABLE_CAIRO
+
 #ifndef _GI_DISABLE_CAIRO
 #include <cairo-gobject.h>
 #endif
-- 
2.7.4

