From 4040e94caf65b6f39787c0db192edacdd981667c Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 4 Nov 2015 15:53:28 -0800
Subject: [PATCH] laird: add laird's version string to EXTRAVERSIOM

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a8c8230..170699e 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@
 VERSION = 2014
 PATCHLEVEL = 07
 SUBLEVEL =
-EXTRAVERSION =
+EXTRAVERSION = -laird01
 NAME =
 
 # *DOCUMENTATION*
-- 
1.9.1

