From ecbe17d34bc56f2dbbc66c3efd0c79996ffb4c67 Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Mon, 17 Feb 2014 17:03:10 +0800
Subject: [PATCH] sama5d3x: detect DM and according it set lcd panel_info

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 board/atmel/sama5d3xek/sama5d3xek.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/board/atmel/sama5d3xek/sama5d3xek.c b/board/atmel/sama5d3xek/sama5d3xek.c
index e0e9d64..ae88092 100644
--- a/board/atmel/sama5d3xek/sama5d3xek.c
+++ b/board/atmel/sama5d3xek/sama5d3xek.c
@@ -152,6 +152,11 @@ void lcd_disable(void)
 
 static void sama5d3xek_lcd_hw_init(void)
 {
+	/* if LCD is PDA7 module, then set output bpp to 18 bit */
+	char dm_id = get_dm_board_id();
+	if (dm_id == BOARD_ID_PDA7_DM)
+		panel_info.vl_bpox = 18;
+
 	gd->fb_base = CONFIG_SAMA5D3_LCD_BASE;
 
 	/* The higher 8 bit of LCD is board related */
-- 
1.9.1

