From f6a91afe0693ef72a0ed232b136ade18f357a65a Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Tue, 26 Aug 2014 17:08:44 +0800
Subject: [PATCH] arm: at91sam9rlek: convert to generic board support

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 include/configs/at91sam9rlek.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/configs/at91sam9rlek.h b/include/configs/at91sam9rlek.h
index 1f493fc..d221ea1 100644
--- a/include/configs/at91sam9rlek.h
+++ b/include/configs/at91sam9rlek.h
@@ -34,6 +34,8 @@
 #define CONFIG_CMD_BOOTZ
 #define CONFIG_OF_LIBFDT
 
+#define CONFIG_SYS_GENERIC_BOARD
+
 #define CONFIG_ATMEL_LEGACY
 #define CONFIG_AT91_GPIO		1
 #define CONFIG_AT91_GPIO_PULLUP		1
-- 
1.9.1

