From 9307220f713dd4ea71bbdfdeed8517fb22285b88 Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Mon, 30 Jun 2014 17:07:46 +0800
Subject: [PATCH] sama5d3_xplained: enable PMECC header generation

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 include/configs/sama5d3_xplained.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/sama5d3_xplained.h b/include/configs/sama5d3_xplained.h
index f72ab0b..723ca1e 100644
--- a/include/configs/sama5d3_xplained.h
+++ b/include/configs/sama5d3_xplained.h
@@ -247,6 +247,7 @@
 #define CONFIG_SYS_NAND_OOBSIZE		64
 #define CONFIG_SYS_NAND_BLOCK_SIZE	0x20000
 #define CONFIG_SYS_NAND_BAD_BLOCK_POS	0x0
+#define CONFIG_SPL_GENERATE_ATMEL_PMECC_HEADER
 
 #endif
 
-- 
1.9.1

