From 0a516c8ad5f67061db652accf36d274c3baa62d4 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 16 Mar 2016 18:18:53 -0700
Subject: [PATCH] laird: update version to laird04

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1b31286..cf57835 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@
 VERSION = 2014
 PATCHLEVEL = 07
 SUBLEVEL =
-EXTRAVERSION = -laird03
+EXTRAVERSION = -laird04
 NAME =
 
 # *DOCUMENTATION*
-- 
1.9.1

