From 655fa279f211264eac8a768a88e7ac7dba3822c1 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Sat, 5 Dec 2015 12:56:35 -0800
Subject: [PATCH] laird: wb50n: upgrade NAND ECC strength to 8 bits

Bug: 8689
---
 include/configs/wb50n.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/wb50n.h b/include/configs/wb50n.h
index bbe12be..fc6d89c 100644
--- a/include/configs/wb50n.h
+++ b/include/configs/wb50n.h
@@ -118,7 +118,7 @@
 /* PMECC & PMERRLOC */
 #define CONFIG_ATMEL_NAND_HWECC
 #define CONFIG_ATMEL_NAND_HW_PMECC
-#define CONFIG_PMECC_CAP		4
+#define CONFIG_PMECC_CAP		8
 #define CONFIG_PMECC_SECTOR_SIZE	512
 #define CONFIG_CMD_NAND_TRIMFFS
 #endif
-- 
1.9.1

