From 0f0278dbff6608e3e2ccfc69d739857575d665d4 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Mon, 24 Nov 2014 17:18:47 +0800
Subject: [PATCH] ARM: atmel: sama5d4 xplained: enable usb ethernet gadget

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 include/configs/sama5d4_xplained.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/configs/sama5d4_xplained.h b/include/configs/sama5d4_xplained.h
index e29ab1b..dee0956 100644
--- a/include/configs/sama5d4_xplained.h
+++ b/include/configs/sama5d4_xplained.h
@@ -122,6 +122,14 @@
 #define CONFIG_USB_STORAGE
 #endif
 
+/* USB device */
+#define CONFIG_USB_GADGET
+#define CONFIG_USB_GADGET_DUALSPEED
+#define CONFIG_USB_GADGET_ATMEL_USBA
+#define CONFIG_USB_ETHER
+#define CONFIG_USB_ETH_RNDIS
+#define CONFIG_USBNET_MANUFACTURER      "Atmel SAMA5D4 Xplained"
+
 #if defined(CONFIG_CMD_USB) || defined(CONFIG_CMD_MMC)
 #define CONFIG_CMD_FAT
 #define CONFIG_DOS_PARTITION
-- 
1.9.1

