From f256423427d3d003c119ab7a7a03f8242b9a21b1 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Sat, 5 Dec 2015 18:43:26 -0800
Subject: [PATCH] laird: update version to laird03

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 735bbce..1b31286 100644
--- a/Makefile
+++ b/Makefile
@@ -8,7 +8,7 @@
 VERSION = 2014
 PATCHLEVEL = 07
 SUBLEVEL =
-EXTRAVERSION = -laird02
+EXTRAVERSION = -laird03
 NAME =
 
 # *DOCUMENTATION*
-- 
1.9.1

