From 90f49eb3506a1eb21b27cdcaf2541ddf0ff17735 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Mon, 24 Nov 2014 17:33:28 +0800
Subject: [PATCH] ARM: atmel: sama5d4ek: add usb device initial code

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 board/atmel/sama5d4ek/sama5d4ek.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/board/atmel/sama5d4ek/sama5d4ek.c b/board/atmel/sama5d4ek/sama5d4ek.c
index daed5cb..a173fe5 100644
--- a/board/atmel/sama5d4ek/sama5d4ek.c
+++ b/board/atmel/sama5d4ek/sama5d4ek.c
@@ -10,6 +10,7 @@
 #include <asm/arch/at91_common.h>
 #include <asm/arch/at91_pmc.h>
 #include <asm/arch/at91_rstc.h>
+#include <asm/arch/atmel_usba_udc.h>
 #include <asm/arch/gpio.h>
 #include <asm/arch/clk.h>
 #include <asm/arch/sama5d3_smc.h>
@@ -106,6 +107,18 @@ static void sama5d4ek_usb_hw_init(void)
 }
 #endif
 
+#ifdef CONFIG_USB_GADGET_ATMEL_USBA
+static void sama5d4ek_udp_hw_init(void)
+{
+	struct at91_pmc *pmc = (struct at91_pmc *)ATMEL_BASE_PMC;
+
+	/* Enable UPLL clock */
+	writel(AT91_PMC_UPLLEN | AT91_PMC_BIASEN, &pmc->uckr);
+	/* Enable UDPHS clock */
+	at91_periph_clk_enable(ATMEL_ID_UDPHS);
+}
+#endif
+
 #ifdef CONFIG_LCD
 vidinfo_t panel_info = {
 	.vl_col = 800,
@@ -305,6 +318,9 @@ int board_init(void)
 #ifdef CONFIG_GENERIC_ATMEL_MCI
 	sama5d4ek_mci1_hw_init();
 #endif
+#ifdef CONFIG_USB_GADGET_ATMEL_USBA
+	sama5d4ek_udp_hw_init();
+#endif
 #ifdef CONFIG_MACB
 	sama5d4ek_macb0_hw_init();
 #endif
-- 
1.9.1

