From 152218ba7b7756512735a7b5fa45c460587b16e1 Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Fri, 29 Aug 2014 10:16:02 +0800
Subject: [PATCH] ARM: at91sam9n12ek: convert to generic board support

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 include/configs/at91sam9n12ek.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/at91sam9n12ek.h b/include/configs/at91sam9n12ek.h
index 9b0e588..f02fce9 100644
--- a/include/configs/at91sam9n12ek.h
+++ b/include/configs/at91sam9n12ek.h
@@ -31,6 +31,7 @@
 #define CONFIG_DISPLAY_CPUINFO
 
 #define CONFIG_OF_LIBFDT
+#define CONFIG_SYS_GENERIC_BOARD
 
 /* general purpose I/O */
 #define CONFIG_AT91_GPIO
-- 
1.9.1

