From e1c6ea2e86bf09228f77ea15fadaf01108a64bdf Mon Sep 17 00:00:00 2001
From: Eric Bentley <eric.bentley@lairdtech.com>
Date: Thu, 15 Jan 2015 16:06:18 -0500
Subject: [PATCH] correct passwd's calculated date to be non-zero

BUG 4543
---
 libbb/update_passwd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libbb/update_passwd.c b/libbb/update_passwd.c
index a30af6f..478aaf7 100644
--- a/libbb/update_passwd.c
+++ b/libbb/update_passwd.c
@@ -227,7 +227,7 @@ int FAST_FUNC update_passwd(const char *filename,
 				cp = strchrnul(cp + 1, ':');
 				/* "name:" + "new_passwd" + ":" + "change date" + ":rest of line" */
 				fprintf(new_fp, "%s%s:%u%s\n", name_colon, new_passwd,
-					(unsigned)(time(NULL)) / (24*60*60), cp);
+					(unsigned)(time(NULL)) / (24*60*60) +1, cp);
 			} else {
 				/* "name:" + "new_passwd" + ":rest of line" */
 				fprintf(new_fp, "%s%s%s\n", name_colon, new_passwd, cp);
-- 
1.9.2

