--- busybox-1.31.1/networking/udhcp/dhcpc.c
+++ busybox-1.31.1/networking/udhcp/dhcpc.c
@@ -1122,6 +1122,7 @@
 	case RENEWING:
 	case REBINDING:
 		client_data.state = RENEW_REQUESTED;
+		udhcp_run_script(NULL, "renewing");
 		break;
 	case RENEW_REQUESTED: /* impatient are we? fine, square 1 */
 		udhcp_run_script(NULL, "deconfig");
@@ -1164,6 +1165,7 @@
 
 	change_listen_mode(LISTEN_NONE);
 	client_data.state = RELEASED;
+	udhcp_run_script(NULL, "released");
 }
 
 static uint8_t* alloc_dhcp_option(int code, const char *str, int extra)
@@ -1523,6 +1525,7 @@
 			case BOUND:
 				/* 1/2 lease passed, enter renewing state */
 				client_data.state = RENEWING;
+				udhcp_run_script(NULL, "renewing");
 				client_data.first_secs = 0; /* make secs field count from 0 */
 				change_listen_mode(LISTEN_KERNEL);
 				log1("entering renew state");
@@ -1560,6 +1563,7 @@
 				/* Timed out or error, enter rebinding state */
 				log1("entering rebinding state");
 				client_data.state = REBINDING;
+				udhcp_run_script(NULL, "rebinding");
 				/* fall right through */
 			case REBINDING:
 				/* Switch to bcast receive */
@@ -1712,6 +1716,7 @@
 
 				/* enter requesting state */
 				client_data.state = REQUESTING;
+				udhcp_run_script(NULL, "requesting");
 				timeout = 0;
 				packet_num = 0;
 				already_waited_sec = 0;
