--- busybox-1.21.1/loginutils/adduser.c	2014-03-25 13:31:53.832582932 -0500
+++ busybox-1.21.1/loginutils/adduser.c	2014-03-25 13:31:53.832582932 -0500
@@ -219,17 +219,17 @@
 	 * 3. last password change (unix date (unix time/24*60*60))
 	 * 4. minimum days required between password changes
 	 * 5. maximum days password is valid
 	 * 6. days before password is to expire that user is warned
 	 * 7. days after password expires that account is disabled
 	 * 8. unix date when login expires (i.e. when it may no longer be used)
 	 */
 	/* fields:     2 3  4 5     6 78 */
-	p = xasprintf("!:%u:0:99999:7:::", (unsigned)(time(NULL)) / (24*60*60));
+	p = xasprintf("!:%u:0:99999:7:::", (unsigned)(time(NULL)) / (24*60*60) + 1);
 	/* ignore errors: if file is missing we suppose admin doesn't want it */
 	update_passwd(bb_path_shadow_file, pw.pw_name, p, NULL);
 	if (ENABLE_FEATURE_CLEAN_UP)
 		free(p);
 #endif
 
 	/* add to group */
 	addgroup_wrapper(&pw, usegroup);
