From 6dcbba80c7efacef57c4154bc88b3002b02be7e3 Mon Sep 17 00:00:00 2001
From: Chris Feyerchak <chris.feyerchak@lairdtech.com>
Date: Tue, 15 Aug 2017 13:05:39 -0400
Subject: [PATCH] Install 'btmgmt' executable

---
 Makefile.tools | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/Makefile.tools b/Makefile.tools
index 7706dc7..7883386 100644
--- a/Makefile.tools
+++ b/Makefile.tools
@@ -343,7 +343,9 @@ EXTRA_DIST += tools/btattach.1
 endif
 
 if READLINE
-noinst_PROGRAMS += attrib/gatttool tools/btmgmt \
+bin_PROGRAMS += tools/btmgmt 
+
+noinst_PROGRAMS += attrib/gatttool \
 			tools/obex-client-tool tools/obex-server-tool \
 			tools/bluetooth-player tools/obexctl
 
-- 
1.9.1

