diff -urN a/arch/arm/mach-at91/Kconfig b/arch/arm/mach-at91/Kconfig
--- a/arch/arm/mach-at91/Kconfig	2012-01-04 18:55:44.000000000 -0500
+++ b/arch/arm/mach-at91/Kconfig	2012-11-25 12:57:00.000000000 -0500
@@ -342,6 +342,26 @@
 	  with 2 SD/MMC Slots. This is the case for AT91SAM9G20-EK rev. C and
 	  onwards.
 
+config MACH_WB40N
+	bool "Laird Connectivity Workgroup Bridge"
+	help
+	  Select this if you are using Laird Connectivity 40N-series
+	  Workgroup Bridge.
+	  <http://www.lairdconnect.com>
+
+choice
+	prompt "Select WB40N board revision"
+	depends on MACH_WB40N
+	default MACH_WB40N_REV3
+
+config MACH_WB40N_REV2
+	bool "Rev2"
+
+config MACH_WB40N_REV3
+	bool "Rev3"
+
+endchoice
+
 config MACH_CPU9G20
 	bool "Eukrea CPU9G20 board"
 	help
@@ -506,6 +526,7 @@
 	  On AT91sam926x chips, or otherwise when using a higher precision
 	  system clock (of at least several MHz), rounding is less of a
 	  problem so it can be safer to use a decimal values like 100.
+	  default y
 
 choice
 	prompt "Select a UART for early kernel messages"
diff -urN a/arch/arm/mach-at91/Makefile b/arch/arm/mach-at91/Makefile
--- a/arch/arm/mach-at91/Makefile	2012-01-04 18:55:44.000000000 -0500
+++ b/arch/arm/mach-at91/Makefile	2012-11-25 12:57:00.000000000 -0500
@@ -62,6 +62,7 @@
 
 # AT91SAM9G20 board-specific support
 obj-$(CONFIG_MACH_AT91SAM9G20EK) += board-sam9g20ek.o
+obj-$(CONFIG_MACH_WB40N)	+= board-wb40n.o
 obj-$(CONFIG_MACH_CPU9G20)	+= board-cpu9krea.o
 obj-$(CONFIG_MACH_ACMENETUSFOXG20) += board-foxg20.o
 obj-$(CONFIG_MACH_STAMP9G20)	+= board-stamp9g20.o
diff -urN a/arch/arm/mach-at91/at91sam9_alt_reset.S b/arch/arm/mach-at91/at91sam9_alt_reset.S
--- a/arch/arm/mach-at91/at91sam9_alt_reset.S	2012-01-04 18:55:44.000000000 -0500
+++ b/arch/arm/mach-at91/at91sam9_alt_reset.S	2012-11-25 12:57:00.000000000 -0500
@@ -31,12 +31,12 @@
 			ldr	r1, .at91_va_base_rstc_cr
 
 			mov	r2, #1
-			mov	r3, #AT91_SDRAMC_LPCB_POWER_DOWN
+			mov	r3, #AT91_SDRAMC_LPCB_DISABLE
 			ldr	r4, =AT91_RSTC_KEY | AT91_RSTC_PERRST | AT91_RSTC_PROCRST
 
 			.balign	32				@ align to cache line
 
-			str	r2, [r0, #AT91_SDRAMC_TR]	@ disable SDRAM access
+@			str	r2, [r0, #AT91_SDRAMC_TR]	@ disable SDRAM access
 			str	r3, [r0, #AT91_SDRAMC_LPR]	@ power down SDRAM
 			str	r4, [r1]			@ reset processor
 
diff -urN a/arch/arm/mach-at91/board-wb40n.c b/arch/arm/mach-at91/board-wb40n.c
--- a/arch/arm/mach-at91/board-wb40n.c	1969-12-31 19:00:00.000000000 -0500
+++ b/arch/arm/mach-at91/board-wb40n.c	2012-11-25 12:57:00.000000000 -0500
@@ -0,0 +1,357 @@
+/*
+ * Copyright (C) 2011-2020 Laird Connectivity, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#include <linux/types.h>
+#include <linux/init.h>
+#include <linux/mm.h>
+#include <linux/module.h>
+#include <linux/platform_device.h>
+#include <linux/clk.h>
+
+#include <mach/hardware.h>
+#include <asm/setup.h>
+#include <asm/mach-types.h>
+#include <asm/irq.h>
+
+#include <asm/mach/arch.h>
+#include <asm/mach/map.h>
+#include <asm/mach/irq.h>
+
+#include <mach/board.h>
+#include <mach/gpio.h>
+#include <linux/gpio_keys.h>
+#include <linux/input.h>
+#include <mach/at91sam9_smc.h>
+#include <mach/system_rev.h>
+
+#include "sam9_smc.h"
+#include "generic.h"
+
+static void __init wb40n_init_early(void)
+{
+	/* Initialize processor: 18.432 MHz crystal */
+	at91_initialize(18432000);
+
+	/* DBGU on ttyS0. (Rx & Tx only) */
+	at91_register_uart(0, 0, 0);
+
+	/* USART0 on ttyS1. (Rx, Tx, CTS, RTS, DTR, DSR, DCD, RI) */
+	at91_register_uart(AT91SAM9260_ID_US0, 1, ATMEL_UART_CTS | ATMEL_UART_RTS
+			   | ATMEL_UART_DTR | ATMEL_UART_DSR | ATMEL_UART_DCD
+			   | ATMEL_UART_RI);
+
+	/* USART1 on ttyS2. (Rx, Tx, RTS, CTS) */
+	at91_register_uart(AT91SAM9260_ID_US1, 2, ATMEL_UART_CTS | ATMEL_UART_RTS);
+
+	/* USART3 on ttyS3 - Bluetooth interface. (Rx, Tx, RTS, CTS) */
+	at91_register_uart(AT91SAM9260_ID_US3, 3, ATMEL_UART_CTS | ATMEL_UART_RTS);
+
+	/* set serial console to ttyS0 (ie, DBGU) */
+	at91_set_serial_console(0);
+}
+
+/*
+ * USB Host port
+ */
+static struct at91_usbh_data __initdata wb40n_usbh_data = {
+	.ports		= 2,
+};
+
+/*
+ * USB Device port
+ */
+static struct at91_udc_data __initdata wb40n_udc_data = {
+	.vbus_pin	= AT91_PIN_PC21,
+	.vbus_active_low = 1,
+	.pullup_pin	= 0, /* pull-up driven by UDC on the AT91SAM9G20 */
+};
+
+/*
+ * Audio (BlueTooth PCM interface)
+ */
+struct ssd40nbt_info {
+	int		ssc_id;
+	struct clk	*dac_clk;
+	char		shortname[32];
+};
+
+static struct ssd40nbt_info ssd40nbt_data = {
+	.ssc_id		= 0,
+	.shortname	= "SSD40NBT Bluetooth Audio Path",
+};
+
+#if defined(CONFIG_SND_SSD40NBT)
+static void __init ssd40nbt_set_clk(struct ssd40nbt_info *info)
+{
+	struct clk *pck0;
+	struct clk *plla;
+
+	pck0 = clk_get(NULL, "pck0");
+	plla = clk_get(NULL, "plla");
+
+	/* SSD40NBT MCK Clock */
+	at91_set_B_periph(AT91_PIN_PB16, 0);	/* PCK0 */
+
+	clk_set_parent(pck0, plla);
+	clk_put(plla);
+
+	info->dac_clk = pck0;
+}
+#else
+static void __init ssd40nbt_set_clk(struct ssd40nbt_info *info) {}
+#endif
+
+/*
+ * MACB Ethernet device
+ */
+static struct at91_eth_data __initdata wb40n_macb_data = {
+	.phy_irq_pin	= AT91_PIN_PB1,
+	.is_rmii	= 1,
+};
+
+static void __init wb40n_add_device_macb(void)
+{
+	at91_add_device_eth(&wb40n_macb_data);
+}
+
+/*
+ * NAND flash
+ */
+static struct mtd_partition __initdata wb40n_nand_partition[] = {
+	{
+		.name   = "at91bs",
+		.offset = 0,
+		.size   = 0x00020000,
+	},
+	{
+		.name	= "u-boot",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x00080000,
+	},
+	{
+		.name	= "u-boot-env-a",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x00020000,
+	},
+	{
+		.name	= "u-boot-env-b",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x00020000,
+	},
+	{
+		.name	= "kernel-a",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x00500000,
+	},
+	{
+		.name	= "kernel-b",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x00500000,
+	},
+	{
+		.name	= "rootfs-a",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x03000000,
+	},
+	{
+		.name	= "rootfs-b",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x03000000,
+	},
+	{
+		.name	= "user",
+		.offset	= MTDPART_OFS_NXTBLK,
+		.size	= 0x014a0000,
+	},
+};
+
+static struct atmel_nand_data __initdata wb40n_nand_data = {
+	.ale		= 21,
+	.cle		= 22,
+//	.det_pin	= ... not connected
+	.rdy_pin	= AT91_PIN_PC13,
+	.enable_pin	= AT91_PIN_PC14,
+	.parts		= wb40n_nand_partition,
+	.num_parts	= ARRAY_SIZE(wb40n_nand_partition),
+};
+
+static struct sam9_smc_config __initdata wb40n_nand_smc_config = {
+	.ncs_read_setup		= 0,
+	.nrd_setup		= 2,
+	.ncs_write_setup	= 0,
+	.nwe_setup		= 2,
+
+	.ncs_read_pulse		= 6,
+	.nrd_pulse		= 3,
+	.ncs_write_pulse	= 6,
+	.nwe_pulse		= 3,
+
+	.read_cycle		= 6,
+	.write_cycle		= 6,
+
+	.mode			= AT91_SMC_READMODE | AT91_SMC_WRITEMODE | \
+				  AT91_SMC_EXNWMODE_DISABLE | AT91_SMC_TDFMODE,
+	.tdf_cycles		= 12,
+};
+
+static void __init wb40n_add_device_nand(void)
+{
+	wb40n_nand_data.bus_width_16 = board_have_nand_16bit();
+	/* setup bus-width (8 or 16) */
+	if (wb40n_nand_data.bus_width_16)
+		wb40n_nand_smc_config.mode |= AT91_SMC_DBW_16;
+	else
+		wb40n_nand_smc_config.mode |= AT91_SMC_DBW_8;
+
+	/* configure chip-select 3 (NAND) */
+	sam9_smc_configure(3, &wb40n_nand_smc_config);
+
+	at91_add_device_nand(&wb40n_nand_data);
+}
+
+/*
+ * MCI (SD/MMC)
+ * wp_pin and vcc_pin are not connected
+ */
+#if defined(CONFIG_MMC_ATMELMCI) || defined(CONFIG_MMC_ATMELMCI_MODULE)
+static struct mci_platform_data __initdata wb40n_mmc_data = {
+	.slot[0] = {
+		.bus_width	= 4,
+		.detect_pin	= AT91_PIN_PC5,
+	},
+	.slot[1] = {
+		.bus_width	= 4,
+//		.detect_pin	= AT91_PIN_PC11,
+	},
+
+};
+#else
+static unsigned int wb40n_slot_b = 0;
+
+static int __init wb40n_slot_b_setup(char *options)
+{
+	if (!strcmp(options, "0"))
+		wb40n_slot_b = 0;
+	else if (!strcmp(options, "no"))
+		wb40n_slot_b = 0;
+	return 0;
+}
+__setup("slot_b=", wb40n_slot_b_setup);
+
+static struct at91_mmc_data __initdata wb40n_mmc_data = {
+	.slot_b		= 0,
+	.wire4		= 1,
+//	.det_pin	= AT91_PIN_PC11,
+};
+#endif
+
+static void __init wb40n_add_device_mmc(void)
+{
+#if defined(CONFIG_MMC_ATMELMCI) || defined(CONFIG_MMC_ATMELMCI_MODULE)
+	at91_add_device_mci(0, &wb40n_mmc_data);
+#else
+	wb40n_mmc_data.slot_b = wb40n_slot_b,
+	at91_add_device_mmc(0, &wb40n_mmc_data);
+#endif
+}
+
+/*
+ * LEDs
+ */
+static struct gpio_led wb40n_leds[] = {
+	{	/* stat0 */
+		.name			= "stat0",
+		.gpio			= AT91_PIN_PA28,
+		.active_low		= 1,
+		.default_trigger	= "none",
+	},
+	{	/* stat1 */
+		.name			= "stat1",
+		.gpio			= AT91_PIN_PA29,
+		.active_low		= 1,
+		.default_trigger	= "none",
+	},
+	{	/* led0 */
+		.name			= "led0",
+		.gpio			= AT91_PIN_PA25,
+		.active_low		= 1,
+		.default_trigger	= "none",
+	},
+	{	/* led1 */
+		.name			= "led1",
+		.gpio			= AT91_PIN_PA26,
+		.default_trigger	= "heartbeat",
+	},
+	{	/* led2 */
+		.name			= "led2",
+		.gpio			= AT91_PIN_PA27,
+		.active_low		= 1,
+		.default_trigger	= "none",
+	}
+};
+
+static void __init wb40n_add_device_gpio_leds(void)
+{
+	at91_gpio_leds(wb40n_leds, ARRAY_SIZE(wb40n_leds));
+}
+
+static void __init wb40n_board_init(void)
+{
+	/* Serial */
+	at91_add_device_serial();
+	/* USB Host */
+#if defined(CONFIG_MACH_WB40N_REV2)
+	at91_set_gpio_output(AT91_PIN_PA22, 0); /* USB current switch enable, low true */
+	at91_set_gpio_input(AT91_PIN_PC15, 1); /* USB current limit (FAULT), low true */
+#else /* REV3 */
+	at91_set_gpio_output(AT91_PIN_PC0, 0); /* USB current switch enable, low true */
+	at91_set_gpio_input(AT91_PIN_PC1, 1); /* USB current limit (FAULT), low true */
+#endif
+	at91_add_device_usbh(&wb40n_usbh_data);
+	/* USB Device */
+	at91_set_gpio_input(AT91_PIN_PC21, 0); /* USB BUS voltage detect, internal pullup disabled */
+	at91_set_gpio_output(AT91_PIN_PC20, 0); /* USB device pullup-enable on H+W board - LEAVE THIS TURNED OFF! */
+	at91_add_device_udc(&wb40n_udc_data);
+	/* NAND */
+	wb40n_add_device_nand();
+	/* Ethernet */
+	wb40n_add_device_macb();
+	/* MMC */
+	wb40n_add_device_mmc();
+	/* SSC (BlueTooth interface of SSD40NBT) */
+	ssd40nbt_set_clk(&ssd40nbt_data);
+	at91_add_device_ssc(AT91SAM9260_ID_SSC, ATMEL_SSC_TX | ATMEL_SSC_RD);
+	/* LEDs */
+	wb40n_add_device_gpio_leds();
+	/* Wifi Module config */
+	/* SYS_RST_L - De-assert system reset */
+	at91_set_gpio_output(AT91_PIN_PB13, 1);
+	/* CHIP_PWD_L - De-assert powerdown */
+	at91_set_gpio_output(AT91_PIN_PB31, 1);
+	/* BT_RST_L - Hold BlueTooth in reset until it is needed by the BT stack */
+	at91_set_gpio_output(AT91_PIN_PC11, 0);
+}
+
+MACHINE_START(WB40N, "Workgroup Bridge 40N")
+	/* Maintainer: support@lairdconnect.com */
+	.timer		= &at91sam926x_timer,
+	.map_io		= at91_map_io,
+	.init_early	= wb40n_init_early,
+	.init_irq	= at91_init_irq_default,
+	.init_machine	= wb40n_board_init,
+MACHINE_END
diff -urN a/drivers/mmc/host/Kconfig b/drivers/mmc/host/Kconfig
--- a/drivers/mmc/host/Kconfig	2012-01-04 18:55:44.000000000 -0500
+++ b/drivers/mmc/host/Kconfig	2012-11-25 12:57:00.000000000 -0500
@@ -297,6 +297,13 @@
 
 endchoice
 
+config MMC_AT91_F_MAX
+	int "Maximum SDIO clock speed"
+	depends on MMC_AT91
+	default 25000000
+	help
+	  This sets the upper limit of the SDIO bus speed.
+
 config MMC_IMX
 	tristate "Motorola i.MX Multimedia Card Interface support"
 	depends on ARCH_MX1
diff -urN a/drivers/mmc/host/at91_mci.c b/drivers/mmc/host/at91_mci.c
--- a/drivers/mmc/host/at91_mci.c	2012-01-04 18:55:44.000000000 -0500
+++ b/drivers/mmc/host/at91_mci.c	2012-11-25 12:57:00.000000000 -0500
@@ -924,6 +924,18 @@
 	.enable_sdio_irq = at91_mci_enable_sdio_irq,
 };
 
+#ifdef CONFIG_MMC_AT91_F_MAX
+static int mmc_at91_f_max = CONFIG_MMC_AT91_F_MAX;
+#else
+static int mmc_at91_f_max = 25000000;
+#endif
+static int __init mmc_at91_f_max_setup(char *options)
+{
+	mmc_at91_f_max = simple_strtol(options, NULL, 0);
+	return 0;
+}
+__setup("sdio_fmax=", mmc_at91_f_max_setup);
+
 /*
  * Probe for the device
  */
@@ -950,7 +962,8 @@
 
 	mmc->ops = &at91_mci_ops;
 	mmc->f_min = 375000;
-	mmc->f_max = 25000000;
+	printk(KERN_INFO "%s: Setting SDIO bus speed to %d\n", pdev->name, mmc_at91_f_max);
+	mmc->f_max = mmc_at91_f_max;
 	mmc->ocr_avail = MMC_VDD_32_33 | MMC_VDD_33_34;
 	mmc->caps = 0;
 
diff -urN a/drivers/tty/n_tty.c b/drivers/tty/n_tty.c
--- a/drivers/tty/n_tty.c	2012-01-04 18:55:44.000000000 -0500
+++ b/drivers/tty/n_tty.c	2012-11-25 12:57:00.000000000 -0500
@@ -50,7 +50,6 @@
 #include <linux/uaccess.h>
 #include <linux/module.h>
 
-#include <asm/system.h>
 
 /* number of characters left in xmit buffer before select has we have room */
 #define WAKEUP_CHARS 256
@@ -1637,6 +1636,7 @@
 	int retval;
 	size_t n;
 	unsigned long flags;
+	bool is_eof;
 
 	retval = 0;
 	spin_lock_irqsave(&tty->read_lock, flags);
@@ -1646,15 +1646,15 @@
 	if (n) {
 		retval = copy_to_user(*b, &tty->read_buf[tty->read_tail], n);
 		n -= retval;
+		is_eof = n == 1 &&
+			tty->read_buf[tty->read_tail] == EOF_CHAR(tty);
 		tty_audit_add_data(tty, &tty->read_buf[tty->read_tail], n);
 		spin_lock_irqsave(&tty->read_lock, flags);
 		tty->read_tail = (tty->read_tail + n) & (N_TTY_BUF_SIZE-1);
 		tty->read_cnt -= n;
 		/* Turn single EOF into zero-length read */
-		if (L_EXTPROC(tty) && tty->icanon && n == 1) {
-			if (!tty->read_cnt && (*b)[n-1] == EOF_CHAR(tty))
-				n--;
-		}
+		if (L_EXTPROC(tty) && tty->icanon && is_eof && !tty->read_cnt)
+			n = 0;
 		spin_unlock_irqrestore(&tty->read_lock, flags);
 		*b += n;
 		*nr -= n;
diff -urN a/net/ipv4/devinet.c b/net/ipv4/devinet.c
--- a/net/ipv4/devinet.c	2012-01-04 18:55:44.000000000 -0500
+++ b/net/ipv4/devinet.c	2012-11-25 12:57:04.000000000 -0500
@@ -1846,4 +1846,4 @@
 	rtnl_register(PF_INET, RTM_DELADDR, inet_rtm_deladdr, NULL, NULL);
 	rtnl_register(PF_INET, RTM_GETADDR, NULL, inet_dump_ifaddr, NULL);
 }
-
+EXPORT_SYMBOL(inet_confirm_addr);
diff -urN a/net/wireless/sme.c b/net/wireless/sme.c
--- a/net/wireless/sme.c	2012-01-04 18:55:44.000000000 -0500
+++ b/net/wireless/sme.c	2012-11-25 12:57:00.000000000 -0500
@@ -470,6 +470,7 @@
 		wdev->conn->state = CFG80211_CONN_IDLE;
 
 	if (status != WLAN_STATUS_SUCCESS) {
+WLAN_NOT_SUCCESSFUL:
 		wdev->sme_state = CFG80211_SME_IDLE;
 		if (wdev->conn)
 			kfree(wdev->conn->ie);
@@ -490,6 +491,18 @@
 				       WLAN_CAPABILITY_ESS,
 				       WLAN_CAPABILITY_ESS);
 
+	if(!bss)
+	{
+/*
+		We occasionally get to this point where cfg80211 thinks we are connected
+		but we have no BSSID.  In order to resync the state machine of cfg80211
+		and the driver, we move to the IDLE state.  This allows the supplicant 
+		to reissue a connect.  Note that we did try calling the device's 
+		disconnect function but that did not help the reconnection speed.
+*/
+		goto WLAN_NOT_SUCCESSFUL;
+	}
+
 	if (WARN_ON(!bss))
 		return;
 
diff -urN a/scripts/setlocalversion b/scripts/setlocalversion
--- a/scripts/setlocalversion	2012-01-04 18:55:44.000000000 -0500
+++ b/scripts/setlocalversion	2012-11-25 15:02:30.000000000 -0500
@@ -170,7 +170,7 @@
 	# LOCALVERSION= is not specified
 	if test "${LOCALVERSION+set}" != "set"; then
 		scm=$(scm_version --short)
-		res="$res${scm:++}"
+#		res="$res${scm:++}"
 	fi
 fi
 
diff -urN a/sound/soc/atmel/Kconfig b/sound/soc/atmel/Kconfig
--- a/sound/soc/atmel/Kconfig	2012-01-04 18:55:44.000000000 -0500
+++ b/sound/soc/atmel/Kconfig	2012-11-25 12:57:00.000000000 -0500
@@ -7,7 +7,7 @@
 	  to select the audio interfaces to support below.
 
 config SND_ATMEL_SOC_SSC
-	tristate
+	tristate "Atmel SSC codec support"
 	depends on SND_ATMEL_SOC
 	help
 	  Say Y or M if you want to add support for codecs the
@@ -24,6 +24,15 @@
 	  Say Y if you want to add support for SoC audio on WM8731-based
 	  AT91sam9g20 evaluation board.
 
+config SND_SSD40NBT
+	tristate "BlueTooth Audio support for Laird Connectivity SSD40NBT"
+	depends on ATMEL_SSC && ARCH_AT91SAM9G20 && SND_ATMEL_SOC && \
+	           AT91_PROGRAMMABLE_CLOCKS
+	select SND_ATMEL_SOC_SSC
+	help
+	  Say Y if you want to add support for Laird Connectivity 
+	  SSD40NBT BlueTooth Audio.
+
 config SND_AT91_SOC_AFEB9260
 	tristate "SoC Audio support for AFEB9260 board"
 	depends on ARCH_AT91 && MACH_AFEB9260 && SND_ATMEL_SOC
diff -urN a/drivers/net/phy/icplus.c b/drivers/net/phy/icplus.c
--- a/drivers/net/phy/icplus.c	2013-04-26 23:20:57.174981268 -0500
+++ b/drivers/net/phy/icplus.c	2013-04-26 23:41:38.219016678 -0500
@@ -30,16 +30,19 @@
 #include <asm/irq.h>
 #include <asm/uaccess.h>
 
-MODULE_DESCRIPTION("ICPlus IP175C/IP101A/IC1001 PHY drivers");
+MODULE_DESCRIPTION("ICPlus IP175C/IP101A/IP101G/IC1001 PHY drivers");
 MODULE_AUTHOR("Michael Barkowski");
 MODULE_LICENSE("GPL");
 
-/* IP101A/IP1001 */
+/* IP101A/G - IP1001 */
 #define IP10XX_SPEC_CTRL_STATUS		16  /* Spec. Control Register */
 #define IP1001_SPEC_CTRL_STATUS_2	20  /* IP1001 Spec. Control Reg 2 */
 #define IP1001_PHASE_SEL_MASK		3 /* IP1001 RX/TXPHASE_SEL */
 #define IP1001_APS_ON			11  /* IP1001 APS Mode  bit */
-#define IP101A_APS_ON			2   /* IP101A APS Mode bit */
+#define IP101A_G_APS_ON			2   /* IP101A/G APS Mode bit */
+#define IP101A_G_IRQ_CONF_STATUS	0x11	/* Conf Info IRQ & Status Reg */
+#define IP101A_G_IRQ_PIN_USED		(1<<15) /* INTR pin used */
+#define IP101A_G_IRQ_DEFAULT		IP101A_G_IRQ_PIN_USED
 
 static int ip175c_config_init(struct phy_device *phydev)
 {
@@ -98,20 +101,24 @@
 
 static int ip1xx_reset(struct phy_device *phydev)
 {
-	int err, bmcr;
+	int bmcr;
 
 	/* Software Reset PHY */
 	bmcr = phy_read(phydev, MII_BMCR);
+	if (bmcr < 0)
+		return bmcr;
 	bmcr |= BMCR_RESET;
-	err = phy_write(phydev, MII_BMCR, bmcr);
-	if (err < 0)
-		return err;
+	bmcr = phy_write(phydev, MII_BMCR, bmcr);
+	if (bmcr < 0)
+		return bmcr;
 
 	do {
 		bmcr = phy_read(phydev, MII_BMCR);
+		if (bmcr < 0)
+			return bmcr;
 	} while (bmcr & BMCR_RESET);
 
-	return err;
+	return 0;
 }
 
 static int ip1001_config_init(struct phy_device *phydev)
@@ -139,7 +146,7 @@
 	return c;
 }
 
-static int ip101a_config_init(struct phy_device *phydev)
+static int ip101a_g_config_init(struct phy_device *phydev)
 {
 	int c;
 
@@ -147,9 +154,14 @@
 	if (c < 0)
 		return c;
 
+	/* INTR pin used: speed/link/duplex will cause an interrupt */
+	c = phy_write(phydev, IP101A_G_IRQ_CONF_STATUS, IP101A_G_IRQ_DEFAULT);
+	if (c < 0)
+		return c;
+
 	/* Enable Auto Power Saving mode */
 	c = phy_read(phydev, IP10XX_SPEC_CTRL_STATUS);
-	c |= IP101A_APS_ON;
+	c |= IP101A_G_APS_ON;
 
 	return phy_write(phydev, IP10XX_SPEC_CTRL_STATUS, c);
 }
@@ -173,6 +185,15 @@
 	return 0;
 }
 
+static int ip101a_g_ack_interrupt(struct phy_device *phydev)
+{
+	int err = phy_read(phydev, IP101A_G_IRQ_CONF_STATUS);
+	if (err < 0)
+		return err;
+
+	return 0;
+}
+
 static struct phy_driver ip175c_driver = {
 	.phy_id		= 0x02430d80,
 	.name		= "ICPlus IP175C",
@@ -202,11 +223,13 @@
 
 static struct phy_driver ip101a_driver = {
 	.phy_id		= 0x02430c54,
-	.name		= "ICPlus IP101A",
+	.name		= "ICPlus IP101A/G",
 	.phy_id_mask	= 0x0ffffff0,
 	.features	= PHY_BASIC_FEATURES | SUPPORTED_Pause |
 			  SUPPORTED_Asym_Pause,
-	.config_init	= &ip101a_config_init,
+	.flags		= PHY_HAS_INTERRUPT,
+	.ack_interrupt	= ip101a_g_ack_interrupt,
+	.config_init	= &ip101a_g_config_init,
 	.config_aneg	= &genphy_config_aneg,
 	.read_status	= &genphy_read_status,
 	.suspend	= genphy_suspend,
@@ -242,6 +265,7 @@
 static struct mdio_device_id __maybe_unused icplus_tbl[] = {
 	{ 0x02430d80, 0x0ffffff0 },
 	{ 0x02430d90, 0x0ffffff0 },
+	{ 0x02430c54, 0x0ffffff0 },
 	{ }
 };
