--- a/net/core/rtnetlink.c
+++ b/net/core/rtnetlink.c
@@ -2076,15 +2076,14 @@
 		int attrlen = nlh->nlmsg_len - NLMSG_ALIGN(min_len);
 		struct rtattr *attr = (void *)nlh + NLMSG_ALIGN(min_len);
 
 		while (RTA_OK(attr, attrlen)) {
 			unsigned int flavor = attr->rta_type & NLA_TYPE_MASK;
 			if (flavor) {
+				if (flavor <= rta_max[sz_idx])
+					rta_buf[flavor-1] = attr;
-				if (flavor > rta_max[sz_idx])
-					return -EINVAL;
-				rta_buf[flavor-1] = attr;
 			}
 			attr = RTA_NEXT(attr, attrlen);
 		}
 	}
 
 	doit = rtnl_get_doit(family, type);
