From 775efbee7a4b2b1630e07825e56af7185c96f7ac Mon Sep 17 00:00:00 2001
From: Cyrille Pitchen <cyrille.pitchen@atmel.com>
Date: Wed, 11 Feb 2015 12:02:56 +0100
Subject: [PATCH] Makefile: update include pathes for autoconf.h

Signed-off-by: Cyrille Pitchen <cyrille.pitchen@atmel.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8e0764d..a494da2 100644
--- a/Makefile
+++ b/Makefile
@@ -215,7 +215,7 @@ NOSTDINC_FLAGS=-nostdinc -isystem $(shell $(CC) -print-file-name=include)
 
 CPPFLAGS=$(NOSTDINC_FLAGS) -ffunction-sections -g -Os -Wall \
 	-fno-stack-protector -fno-common \
-	-I$(INCL) -Iinclude -Ifs/include \
+	-I$(INCL) -Iinclude -Ifs/include -I$(TOPDIR)/config/at91bootstrap-config \
 	-DAT91BOOTSTRAP_VERSION=\"$(VERSION)$(REV)$(SCMINFO)\" -DCOMPILE_TIME="\"$(DATE)\""
 
 ASFLAGS=-g -Os -Wall -I$(INCL) -Iinclude
-- 
1.9.1

