From e6033d1206316342aa8086b2fa6b2737a2116df8 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 24 Mar 2016 16:35:20 -0700
Subject: [PATCH] Update version to laird08

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index bf7a8cd..9e91d56 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ BINDIR:=$(TOPDIR)/binaries
 
 DATE := $(shell date)
 VERSION := 3.7.1
-REVISION := laird07
+REVISION := laird08
 SCMINFO := $(shell ($(TOPDIR)/host-utilities/setlocalversion $(TOPDIR)))
 
 ifeq ($(SCMINFO),)
-- 
1.9.1

