From f41a2147e2b2c3590e99ce2363a2abeafbe63d18 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 11 Feb 2016 09:42:19 -0800
Subject: [PATCH] laird: update version to laird06

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 625b1d0..7322064 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ BINDIR:=$(TOPDIR)/binaries
 
 DATE := $(shell date)
 VERSION := 3.7.1
-REVISION := laird05
+REVISION := laird06
 SCMINFO := $(shell ($(TOPDIR)/host-utilities/setlocalversion $(TOPDIR)))
 
 ifeq ($(SCMINFO),)
-- 
1.9.1

