From a97e4cbccc7d4fc569ebf72c17384052d67ba405 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Fri, 17 Oct 2014 16:47:43 +0800
Subject: [PATCH] board: sama5d3xek: add ALLOW_FLASH option

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 board/sama5d3xek/Config.in.board | 1 +
 1 file changed, 1 insertion(+)

diff --git a/board/sama5d3xek/Config.in.board b/board/sama5d3xek/Config.in.board
index 69fb810..cd783be 100644
--- a/board/sama5d3xek/Config.in.board
+++ b/board/sama5d3xek/Config.in.board
@@ -3,6 +3,7 @@ config CONFIG_SAMA5D3XEK
 	select SAMA5D3X
 	select CONFIG_DDRC
 	select ALLOW_DATAFLASH
+	select ALLOW_FLASH
 	select ALLOW_NANDFLASH
 	select ALLOW_SDCARD
 	select ALLOW_HSMCI
-- 
1.9.1

