From 5ebb4f5f7def9026ebf3f1a1d615c8983d5dbd49 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Sat, 5 Dec 2015 18:43:51 -0800
Subject: [PATCH] laird: update version to laird04

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8e5a44d..03724d2 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ BINDIR:=$(TOPDIR)/binaries
 
 DATE := $(shell date)
 VERSION := 3.7.1
-REVISION := laird03
+REVISION := laird04
 SCMINFO := $(shell ($(TOPDIR)/host-utilities/setlocalversion $(TOPDIR)))
 
 ifeq ($(SCMINFO),)
-- 
1.9.1

