From efddcf0f9d9a4b034069882514aedaeadf563213 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Fri, 17 Oct 2014 16:46:59 +0800
Subject: [PATCH] driver: makefile: assemble also need FLASH flags

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 driver/driver_cpp.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/driver/driver_cpp.mk b/driver/driver_cpp.mk
index 4b74b5a..29e14f0 100644
--- a/driver/driver_cpp.mk
+++ b/driver/driver_cpp.mk
@@ -72,6 +72,7 @@ endif
 
 ifeq ($(CONFIG_FLASH),y)
 CPPFLAGS += -DCONFIG_FLASH
+ASFLAGS += -DCONFIG_FLASH
 endif
 
 ifeq ($(CONFIG_LOAD_LINUX),y)
-- 
1.9.1

