From d564e64e5284523735cc1f15a4e9f43ea1bf16fb Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Wed, 6 Jan 2016 13:07:34 -0800
Subject: [PATCH] laird: update version to laird05

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 03724d2..625b1d0 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ BINDIR:=$(TOPDIR)/binaries
 
 DATE := $(shell date)
 VERSION := 3.7.1
-REVISION := laird04
+REVISION := laird05
 SCMINFO := $(shell ($(TOPDIR)/host-utilities/setlocalversion $(TOPDIR)))
 
 ifeq ($(SCMINFO),)
-- 
1.9.1

