From c692195893a1cec7d360939774066d6e8f784ab7 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Thu, 3 Mar 2016 17:21:23 -0800
Subject: [PATCH] laird: update version to laird07

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7322064..bf7a8cd 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,7 @@ BINDIR:=$(TOPDIR)/binaries
 
 DATE := $(shell date)
 VERSION := 3.7.1
-REVISION := laird06
+REVISION := laird07
 SCMINFO := $(shell ($(TOPDIR)/host-utilities/setlocalversion $(TOPDIR)))
 
 ifeq ($(SCMINFO),)
-- 
1.9.1

