From 185ca2ed71ff2712cb20d194d2e38c824a4f1f50 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Fri, 17 Oct 2014 16:48:27 +0800
Subject: [PATCH] main: enable to init flash load function

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 main.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/main.c b/main.c
index 93ac180..f91dee6 100644
--- a/main.c
+++ b/main.c
@@ -54,6 +54,8 @@ static int init_loadfunction(void)
 #else
 #if defined (CONFIG_DATAFLASH)
 	load_image = &load_dataflash;
+#elif defined(CONFIG_FLASH)
+	load_image = &load_norflash;
 #elif defined(CONFIG_NANDFLASH)
 	load_image = &load_nandflash;
 #elif defined(CONFIG_SDCARD)
@@ -97,6 +99,17 @@ int main(void)
 	image.of_dest = (unsigned char *)OF_ADDRESS;
 #endif
 
+#ifdef CONFIG_FLASH
+	media_str = "FLASH: ";
+	image.offset = IMG_ADDRESS;
+#if !defined(CONFIG_LOAD_LINUX) && !defined(CONFIG_LOAD_ANDROID)
+	image.length = IMG_SIZE;
+#endif
+#ifdef CONFIG_OF_LIBFDT
+	image.of_offset = OF_OFFSET;
+#endif
+#endif
+
 #ifdef CONFIG_NANDFLASH
 	media_str = "NAND: ";
 	image.offset = IMG_ADDRESS;
-- 
1.9.1

