From dbcc14c258e523345a0b5ee7263ddf7e7dae7a45 Mon Sep 17 00:00:00 2001
From: Steve deRosier <steve.derosier@lairdtech.com>
Date: Mon, 8 Feb 2016 17:05:41 -0800
Subject: [PATCH] laird: wb50: Fix compiler warnings caused by memtest

Bug: 9007
---
 main.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/main.c b/main.c
index 51a1e4f..d2e5601 100644
--- a/main.c
+++ b/main.c
@@ -90,8 +90,6 @@ int main(void)
 	struct image_info image;
 	char *media_str = NULL;
 	int ret;
-	void * retptr;
-	int i;
 
 	char filename[FILENAME_BUF_LEN];
 	char of_filename[FILENAME_BUF_LEN];
@@ -174,6 +172,9 @@ int main(void)
 #endif
 
 #if defined(CONFIG_MEMTEST)
+	void * retptr;
+	int i;
+
 	dbg_info("Doing memtest on load area %d, size: %d\n", CONFIG_MEMTEST_START, CONFIG_MEMTEST_SIZE);
 
 	for(i = 0; i < CONFIG_MEMTEST_ITTERATIONS; i++)
-- 
1.9.1

