From 66d512008ea7b1c5b0e0f8674e71af11b015ac37 Mon Sep 17 00:00:00 2001
From: Wenyou Yang <wenyou.yang@atmel.com>
Date: Wed, 24 Dec 2014 15:45:32 +0800
Subject: [PATCH] board: sama5d3xek: fix compile ASFLAG error

Signed-off-by: Wenyou Yang <wenyou.yang@atmel.com>
---
 board/sama5d3xek/board.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/board/sama5d3xek/board.mk b/board/sama5d3xek/board.mk
index 15db59f..8e31c58 100644
--- a/board/sama5d3xek/board.mk
+++ b/board/sama5d3xek/board.mk
@@ -8,9 +8,10 @@ ASFLAGS += \
 else
 CPPFLAGS += \
 	-DCONFIG_SAMA5D3XEK \
+	-mcpu=cortex-a5 \
 	-mtune=cortex-a5
 
 ASFLAGS += \
-	-DCONFIG_SAMA5D3XEK
+	-DCONFIG_SAMA5D3XEK \
 	-mcpu=cortex-a5
 endif
-- 
1.9.1

