diff -uar openssl-1.0.2k-orig/ssl/s3_both.c openssl-1.0.2k-new/ssl/s3_both.c
--- openssl-1.0.2k-orig/ssl/s3_both.c	2017-01-26 08:22:04.000000000 -0500
+++ openssl-1.0.2k-new/ssl/s3_both.c	2017-07-14 09:34:34.478062471 -0400
@@ -536,8 +536,6 @@
     case X509_V_ERR_ERROR_IN_CERT_NOT_AFTER_FIELD:
     case X509_V_ERR_ERROR_IN_CRL_LAST_UPDATE_FIELD:
     case X509_V_ERR_ERROR_IN_CRL_NEXT_UPDATE_FIELD:
-    case X509_V_ERR_CERT_NOT_YET_VALID:
-    case X509_V_ERR_CRL_NOT_YET_VALID:
     case X509_V_ERR_CERT_UNTRUSTED:
     case X509_V_ERR_CERT_REJECTED:
     case X509_V_ERR_HOSTNAME_MISMATCH:
@@ -549,6 +547,8 @@
     case X509_V_ERR_CRL_SIGNATURE_FAILURE:
         al = SSL_AD_DECRYPT_ERROR;
         break;
+    case X509_V_ERR_CERT_NOT_YET_VALID:
+    case X509_V_ERR_CRL_NOT_YET_VALID:
     case X509_V_ERR_CERT_HAS_EXPIRED:
     case X509_V_ERR_CRL_HAS_EXPIRED:
         al = SSL_AD_CERTIFICATE_EXPIRED;
Only in openssl-1.0.2k-new/ssl: s3_both.c~
