From 9e1f83500aa502fdc3a42ad0bd560bf7c1909298 Mon Sep 17 00:00:00 2001
From: Mark Adler <madler@alumni.caltech.edu>
Date: Sun, 9 Feb 2020 20:51:43 -0800
Subject: [PATCH] Add prototypes for the cover functions.

---
 extract.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/extract.c b/extract.c
index 0cb7bfc..504afd6 100644
--- a/extract.c
+++ b/extract.c
@@ -340,6 +340,10 @@ typedef struct {
     size_t max;         /* allocated number of spans (num <= max) */
 } cover_t;

+static size_t cover_find OF((cover_t *, bound_t));
+static int cover_within OF((cover_t *, bound_t));
+static int cover_add OF((cover_t *, bound_t, bound_t));
+
 /*
  * Return the index of the first span in cover whose beg is greater than val.
  * If there is no such span, then cover->num is returned.

