From d3934d015d0ce99d3588a3c97316e99fd9620ced Mon Sep 17 00:00:00 2001
From: Mark Adler <madler@alumni.caltech.edu>
Date: Sun, 9 Feb 2020 21:36:39 -0800
Subject: [PATCH] Fix code to compile for USE_ZLIB and old zlib version.

---
 globals.h | 2 +-
 inflate.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/globals.h b/globals.h
index 2bdcdeb..f9c6daf 100644
--- a/globals.h
+++ b/globals.h
@@ -236,6 +236,7 @@ typedef struct Globals {
     uch       *inbuf;               /* input buffer (any size is OK) */
     uch       *inptr;               /* pointer into input buffer */
     int       incnt;
+    int      disk_full;

 #ifndef FUNZIP
     ulg       bitbuf;
@@ -266,7 +267,6 @@ typedef struct Globals {
     uch      *outbufptr;           /* extract.c static */
     ulg      outsize;              /* extract.c static */
     int      reported_backslash;   /* extract.c static */
-    int      disk_full;
     int      newfile;
     void     **cover;              /* used in extract.c for bomb detection */

diff --git a/inflate.c b/inflate.c
index f2f6864..2f5a015 100644
--- a/inflate.c
+++ b/inflate.c
@@ -677,7 +677,7 @@ int UZinflate(__G__ is_defl64)
             if ((!repeated_buf_err) && (G.dstrm.avail_in == 0)) {
                 /* when detecting this problem for the first time,
                    try to provide one fake byte beyond "EOF"... */
-                G.dstrm.next_in = "";
+                G.dstrm.next_in = (Bytef *)"";
                 G.dstrm.avail_in = 1;
                 repeated_buf_err = TRUE;
             } else

