From 1860ba704db791db940475b1fb6ef73bdb81bcab Mon Sep 17 00:00:00 2001
From: Mark Adler <madler@alumni.caltech.edu>
Date: Sun, 9 Feb 2020 21:50:03 -0800
Subject: [PATCH] If ZIP64_SUPPORT is requested, then request
 LARGE_FILE_SUPPORT.

---
 unzpriv.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/unzpriv.h b/unzpriv.h
index 297b3c7..09f288e 100644
--- a/unzpriv.h
+++ b/unzpriv.h
@@ -678,6 +678,9 @@

 #define UNZIP_BZ2VERS   46
 #ifdef ZIP64_SUPPORT
+# ifndef LARGE_FILE_SUPPORT
+#  define LARGE_FILE_SUPPORT
+# endif
 # ifdef USE_BZIP2
 #  define UNZIP_VERSION   UNZIP_BZ2VERS
 # else

