--- a/src/libcrypto.c	2022-09-07 09:29:03.000000000 
+++ b/src/libcrypto.c	2022-09-11 02:47:34.000000000 
@@ -73,7 +73,7 @@
 
 #include "libssh/crypto.h"
 
-#ifdef HAVE_OPENSSL_EVP_KDF_CTX
+#if defined(HAVE_OPENSSL_EVP_KDF_CTX) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
 #include <openssl/kdf.h>
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
 #include <openssl/param_build.h>
@@ -179,7 +179,7 @@
 }
 #endif /* HAVE_OPENSSL_ECC */
 
-#ifdef HAVE_OPENSSL_EVP_KDF_CTX
+#if defined(HAVE_OPENSSL_EVP_KDF_CTX) && (OPENSSL_VERSION_NUMBER >= 0x10100000L)
 #if OPENSSL_VERSION_NUMBER < 0x30000000L
 static const EVP_MD *sshkdf_digest_to_md(enum ssh_kdf_digest digest_type)
 {
