From b344e4ec2a5143dfc4d1aa5954dc947e19a8fb25 Mon Sep 17 00:00:00 2001
From: Jon Hefling <jon.hefling@lairdtech.com>
Date: Tue, 10 Feb 2015 16:16:34 -0600
Subject: [PATCH] Laird: Update version to laird02

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8232ec4..4956969 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 VERSION = 3
 PATCHLEVEL = 2
 SUBLEVEL = 0
-EXTRAVERSION = -laird01
+EXTRAVERSION = -laird02
 NAME = Saber-toothed Squirrel
 
 # *DOCUMENTATION*
-- 
2.1.0

