--- a/src/uboot_env.c	2022-03-28 18:27:35.000000000 
+++ b/src/uboot_env.c	2022-03-29 01:29:24.000000000 
@@ -32,7 +32,6 @@
 #include <sys/stat.h>
 #include <sys/wait.h>
 #include <sys/ioctl.h>
-#include <zlib.h>
 #include <mtd/mtd-user.h>
 #include <mtd/ubi-user.h>
 
@@ -933,6 +932,19 @@
 	return ret;
 }
 
+static uint32_t crc32(uint32_t crc, const uint8_t *buf, size_t len)
+{
+    int k;
+
+    crc = ~crc;
+    while (len--) {
+        crc ^= *buf++;
+        for (k = 0; k < 8; k++)
+            crc = crc & 1 ? (crc >> 1) ^ 0xedb88320 : crc >> 1;
+    }
+    return ~crc;
+}
+
 int libuboot_env_store(struct uboot_ctx *ctx)
 {
 	struct var_entry *entry;
--- a/src/CMakeLists.txt	2022-03-29 01:36:47.000000000 
+++ b/src/CMakeLists.txt	2022-03-29 12:21:41.000000000 
@@ -20,7 +20,7 @@
 
 add_executable(fw_printenv fw_printenv.c)
 add_executable(fw_setenv fw_setenv.c)
-target_link_libraries(ubootenv z)
+target_link_libraries(ubootenv)
 target_link_libraries(fw_printenv ubootenv)
 target_link_libraries(fw_setenv ubootenv)
 
--- a/src/libubootenv.pc.in	2021-11-02 11:11:40.000000000 
+++ b/src/libubootenv.pc.in	2022-08-26 12:36:07.000000000 
@@ -10,5 +10,5 @@
 Name: libubootenv
 Description: Library to access U-Boot environment
 Version: @VERSION@
-Libs: -L${libdir} -lubootenv -lz
+Libs: -L${libdir} -lubootenv
 Cflags: -I${includedir}
